src/network/networkd-routing-policy-rule.c

```diff
@@ -1049,8 +1061,28 @@ int manager_rtnl_process_rule(sd_netlink *rtnl, sd_netlink_message *message, Man
                 log_warning_errno(r, "rtnl: could not get FRA_SUPPRESS_PREFIXLEN attribute, ignoring: %m");
                 return 0;
         }
-        if (r >= 0)
-                tmp->suppress_prefixlen = (int) suppress_prefixlen;
+        if (r >= 0) {
+                // kernel does not limit this value, but we do
+                if (suppress_prefixlen > 128) {
```

Comment:
I've checked kernel sources and it doesn't limit this value in any way, so in theory possible such situation that some buggy tool will set it over `128`.